src/languages/php.js

```diff
@@ -263,13 +262,65 @@ export default function(hljs) {
     "stdClass"
   ];
 
+  /** Dual-case keywords
+   *
+   * ["then","FILE"] =>
+   *     ["then", "THEN", "FILE", "file"]
+   *
+   * @param {string[]} items */
+  const dualCase = (items) => {
+    /** @type string[] */
+    const result = [];
+    items.forEach(item => {
+      result.push(item);
+      if (item.toLowerCase() === item) {
+        result.push(item.toUpperCase());
+      } else {
+        result.push(item.toLowerCase());
+      }
+    });
+    return result;
+  };
+
   const KEYWORDS = {
-    keyword: KWS,
-    literal: LITERALS,
-    built_in: BUILT_INS
+    keyword: KWS.concat([ "match|0" ]),
```

Comment:
No, this is not the right solution.  The keyword lists should be what they are.  If the `|0` tagging causes a problem for your code on line 312 (or elsewhere) then you need to write a filter function that returns just raw keywords by stripping the `|0`...

Reply:
But as I mention elsewhere I also question the need to use the full lists as a negative filter...